test_app/views.py

```diff
@@ -1,3 +1,179 @@
+# -*- coding: utf-8 -*-
+
+import pandas as pd
+import requests
+import json
+import string
+import random
+
 from django.shortcuts import render
+from django.http import HttpResponse
+from django.http import JsonResponse
+from faker import Faker
+
+from . import tools as t
+
+
+def index(request):
+    return HttpResponse("Hello, world!! Select: <br>1. <a href='http://127.0.0.1:5000/get_profile'>Get profile</a><br>" +
+                        "2. <a href='http://127.0.0.1:5000/get_libs'>Get libs</a><br>" +
+                        "3. <a href='http://127.0.0.1:5000/get_avg_wh'>Get average height and weight</a>")
+
+
+def get_profile(request):
+    """Get name and mail from profile"""
+
+    fake = Faker(['it_IT', 'en_US', 'uk_UA'])
+
+    profile_list = [str(fake.profile(fields=['name', 'mail'])) for _ in range(100)]
+    return JsonResponse({"profiles": profile_list})
+
+
+def get_lib(request):
+    """Get libraries in requirements.txt"""
+
+    libs_list = [line.strip() for line in open('requirements.txt', 'r')]
+    return JsonResponse({"libs": libs_list})
+
+
+def get_avg_wh(request):
+
+    """Get student's average height and weight """
+
+    hw_df = pd.read_csv('datasets/hw.csv', delimiter=',')
+    hw_df.columns = ['index', 'height', 'weight']
+
+    avg_height = int(round(hw_df["height"].mean()))
+    avg_weight = int(round(hw_df["weight"].mean()))
+    return JsonResponse({"average": {"height": avg_height,
+                                          "weight": avg_weight
+                                          }})
+
+
+def get_astronauts(request):
+
+    """Get astronauts"""
+
+    response = requests.get("http://api.open-notify.org/astros.json")
+    if response.status_code == 200:
+        resp = json.loads(response.text)
+        return HttpResponse(f"Astronauts number: {resp['number']}")
+    else:
+        return HttpResponse(f"Error {response.status_code}")
+
+
+def get_password(request):
+
+    """
+
+    Password generator that getting two GET parameters: length(length of password)
+    and isdigit(bool param that indicate if number exists in password or not).
+    Length and isdigit must be numeric and greater than zero.
+    Length ranges from 8 to 24.
+
+    Example: length=24&isdigit=0
+
+    """
+
+    length = request.GET.get('length')
+    isdigit = request.GET.get('isdigit')
+
+    if length is not None and isdigit is not None:
+        is_len_digit = length.isdigit()
+        is_dig_digit = isdigit.isdigit()
+
+        if is_dig_digit and is_len_digit:
+            length = int(length)
+            isdigit = int(isdigit)
+            if isdigit not in range(0, 2):
+                return HttpResponse('Value isdigit must be 0 or 1')
+            if length in range(8, 25):
+                generation_symbols = string.ascii_lowercase
+                if isdigit == 1:
+                    generation_symbols += string.digits
+                elif isdigit == 0:
+                    generation_symbols = generation_symbols
```

Comment:
is it needed?

Reply:
Based on the task, it is necessary that the parameter be equal to either 0 or 1.

Reply:
My point was that this assignment `x = x` does nothing and could potentially be skipped

Reply:
Fixed.

> My point was that this assignment `x = x` does nothing and could potentially be skipped